src/parsers/gene_association_file.py

```diff
@@ -0,0 +1,117 @@
+import pandas as pd
+import click 
+
+def process_go_annotations(input_path: str, output_path: str):
+    """
+    Processes GO annotations by cleaning, transforming, and mapping evidence types.
+
+    Args:
+        input_path (str): Path to the input GO annotations CSV file.
+        output_path (str): Path to save the normalized output CSV file.
+    """
+    
+    # load the input data 
+    df = pd.read_csv(input_path)
+
+    # Drop unnecessary columns to simplify the DataFrame
+    drop_columns = [
+        "DB_Object_Symbol", "Aspect", "DB_Object_Name", "Synonym", 
+        "DB_Object_Type", "Taxon", "Annotation_Extension", "Gene_Product_Form_ID"
+        ]
+    
+    df = df.drop(columns = drop_columns)
+
+    # Create a unified 'subject' identifier
+    df["subject"] = df["DB"] + ":" + df["DB_Object_ID"]
+    df = df.drop(columns=["DB", "DB_Object_ID"])
+
+    # Reorder columns to place 'subject' first
+    cols = ["subject"] + [col for col in df.columns if col != "subject"]
+    df = df[cols]
+    
+    # Rename columns to match the desired schema
+    df = df.rename(columns={
+        "Qualifier": "predicate", 
+        "GO_ID": "object",
+        "DB_Reference": "publications", 
+        "With_From": "supporting_objects",
```

Comment:
These two columns are going to need closer attention -- if you look at the specs, [the DB_Reference column](https://geneontology.org/docs/go-annotation-file-gaf-format-2.2/#dbreference-column-6) can contain one or more values, and [the With_From column](https://geneontology.org/docs/go-annotation-file-gaf-format-2.2/#with-or-from-column-8) can contain zero or more values.

Reading the docs more closely, it actually looks like we may need to alter the schema a bit here to support both these columns having more than one entry. Let's discuss later on how we're going to represent this data as it's more complicated than we thought when we were putting the schema together!